xonsh/platform.py

```diff
@@ -0,0 +1,79 @@
+""" Module for platform-specific constants and implementations """
+
+import os
+import platform
+import sys
+
+
+#
+# Python
+#
+
+PYTHON_VERSION = sys.version_info[:3]
+
+if PYTHON_VERSION < (3, 5, 0):
+    def scandir(path):
+        """ Compatibility wrapper for os.scandir from Python 3.5+ """
+        return iter(os.listdir(path))
+else:
+    from os import scandir
+
+
+ON_ANACONDA = any(s in sys.version for s in ('Anaconda', 'Continuum'))
+
+
+#
+# OS
+#
+
+ON_DARWIN = ON_LINUX = ON_WINDOWS = None
+for name in ('Darwin', 'Linux', 'Windows'):
+    globals()['ON_' + name.upper()] = platform.system() == name
+
+
+#
+# Linux distro
+#
+
+if ON_LINUX and '-ARCH-' in platform.platform():
```

Comment:
I would prefer if all of the `ON_*` variables were bools and then if arch or version info can be stored in other variable names.